src/app/favorites/[slug]/page.tsx

```diff
@@ -0,0 +1,76 @@
+'use client';
+import { useState } from 'react';
+import { Button } from '@/components/ButtonStyle';
+import { addToFavorites } from '@/functions/addToFavorites';
+import { getCategory } from '@/functions/getCategory';
+import { normalizeImage } from '@/functions/normalizeImage';
+import { parseURL } from '@/functions/parseUrl';
+import { ItemClothes } from '@/types/itemClothes';
+import { useDispatch, useSelector } from 'react-redux';
+import { AiOutlineHeart } from 'react-icons/ai';
+import {
+  Card,
+  CardFooter,
+  CardHeader,
+  Image,
+  Tooltip
+} from '@nextui-org/react';
+
+const Page = async ({ params }) => {
+  const decodedItem: ItemClothes = parseURL(params);
+  const formattedPicture = normalizeImage(decodedItem.image);
+  const dispatch = useDispatch();
+
+  const { category } = getCategory(decodedItem.category);
+
+  const [favorite, setFavorite] = useState(false);
+
+  const handleToggleFavorite = () => {
+    dispatch(addToFavorites(decodedItem));
+    setFavorite(!favorite);
+  };
+
+  const tooltipContent = !favorite
+    ? 'Remover dos favoritos'
+    : 'Adicionar aos favoritos';
```

Comment:
@samuelrms this can be wrapped within a useMemo to prevent your variable to be rendered when every fragment of the page gets rehydrated.

Reply:
@vaporwavie useMemo added, according to the tip!